### PR TITLE
New package: py-tensorpack

### PR DESCRIPTION
**Pull request by qwertos**

### CODE REVIEW

File: var/spack/repos/builtin/packages/py-tensorpack/package.py [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+    on TensorFlow."""
+
+    homepage = "https://github.com/tensorpack/tensorpack"
+    pypi = "tensorpack/tensorpack-0.10.1.tar.gz"
+
+    version('0.10.1', sha256='ae6af59794459de910725d268061f0c86d78f01948f9fd5d7b11dd9770ad71ef')
+    version('0.9.8', sha256='bc6566c12471a0f9c0a79acc3d045595b1943af8e423c5b843986b73bfe5425f')
+
+    depends_on('python@3.3:', when='@0.10.1:', type=('build', 'run'))
+    depends_on('py-setuptools', type='build')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-tensorpack/package.py]:
```suggestion
    depends_on('py-setuptools@31:', type='build')
```

**Review by adamjstewart — DISMISSED** [review]
@aweits may be interested in this as well

**Comment by adamjstewart** [comment]
Need to add a `py-msgpack-numpy` package

File: var/spack/repos/builtin/packages/py-tensorpack/package.py [hunk 2]
```diff
@@ -0,0 +1,31 @@
+# Copyright 2013-2020 Lawrence Livermore National Security, LLC and other
```
Comment (tldahlgren) [inline var/spack/repos/builtin/packages/py-tensorpack/package.py]:
```suggestion
# Copyright 2013-2021 Lawrence Livermore National Security, LLC and other
```

File: var/spack/repos/builtin/packages/py-tensorpack/package.py [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+
+
+class PyTensorpack(PythonPackage):
+    """Tensorpack is a neural network training interface based
+    on TensorFlow."""
+
+    homepage = "https://github.com/tensorpack/tensorpack"
+    pypi = "tensorpack/tensorpack-0.10.1.tar.gz"
+
+    version('0.10.1', sha256='ae6af59794459de910725d268061f0c86d78f01948f9fd5d7b11dd9770ad71ef')
```
Comment (tldahlgren) [inline var/spack/repos/builtin/packages/py-tensorpack/package.py:18]: Would you be willing to add yourself as the Spack package manager for this package?

**Review by tldahlgren — CHANGES_REQUESTED** [review]

File: var/spack/repos/builtin/packages/py-tensorpack/package.py (hunk 3 above)
Comment (qwertos) [reply var/spack/repos/builtin/packages/py-tensorpack/package.py:18]: I don't know if I'm the best person to list as maintainer for any of the packages I push up. My interaction with actual software that gets installed is VERY minimal. Once our users are happy and the package gets accepted here, I move on to the next package and forget most of the specifics about the last.

Comment (adamjstewart) [reply var/spack/repos/builtin/packages/py-tensorpack/package.py:18]: @aweits may be interested in being a maintainer for this

File: var/spack/repos/builtin/packages/py-tensorpack/package.py [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+    depends_on("py-six", type=('build', 'run'))
+    depends_on("py-termcolor@1.1:", type=('build', 'run'))
+    depends_on("py-tabulate@0.7.7:", type=('build', 'run'))
+    depends_on("py-tqdm@4.29.0.1:", type=('build', 'run'))
+    depends_on("py-msgpack@0.5.2:", type=('build', 'run'))
+    depends_on("py-msgpack-numpy@0.4.4.2:", type=('build', 'run'))
+    depends_on("py-pyzmq@16:", type=('build', 'run'))
+    depends_on("py-psutil@5:", type=('build', 'run'))
+    depends_on('py-subprocess32', when='@:0.9.8 ^python@:2.999', type=('build', 'run'))
+    depends_on('py-functools32',  when='@:0.9.8 ^python@:2.999', type=('build', 'run'))
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-tensorpack/package.py:33]: The current release you added only supports Python 3, so these deps can be removed. Also, according to the README, there are some additional strongly recommended dependencies:

+ Python 3.3+.
+ Python bindings for OpenCV. (Optional, but required by a lot of features)
+ TensorFlow ≥ 1.5, < 2
  * TF is not not required if you only want to use `tensorpack.dataflow` alone as a data processing library
  * TF2 is supported if used in graph mode (and use `tf.compat.v1` when needed)

File: var/spack/repos/builtin/packages/py-tensorpack/package.py (hunk 3 above)
Comment (aweits) [reply var/spack/repos/builtin/packages/py-tensorpack/package.py:18]: @qwertos you can put me as maintainer...

File: var/spack/repos/builtin/packages/py-tensorpack/package.py (hunk 4 above)
Comment (qwertos) [reply var/spack/repos/builtin/packages/py-tensorpack/package.py:31]: > 
> The current release you added only supports Python 3, so these deps can be removed. 
> 

I'm confused. Why would I remove those when they're required for the older version that is still listed?

Comment (alalazo) [reply var/spack/repos/builtin/packages/py-tensorpack/package.py:33]: @qwertos Can you address this? :point_up:

**Comment by alalazo** [comment]
Closing, since there has been no reply in a while. Feel free to reopen if you want to continue work on this.

**Pull request closed without merging**